pallene/checker.lua

```diff
@@ -1137,8 +1137,8 @@ function Checker:check_exp_verify(exp, expected_type, errmsg_fmt, ...)
         -- These assertions are always true in the current version of Pallene, which does not allow
         -- nested function expressions. Once we add function expressions to the parser then we
         -- should convert these assertions into proper calls to type_error.
-        assert(expected_type._tag == "types.T.Function")
-        assert(#expected_type.arg_types == #exp.arg_decls)
+        assert(expected_type._tag == "types.T.Function" and #expected_type.arg_types == #exp.arg_decls,
+            "function expressions not implemented yet")
```

Comment:
We prefer putting each check on a separate assertion, because if the assertion fails then the error message tells which line it was.